### PR TITLE
WWST-2052 LUMI water leak sensor

### PR DESCRIPTION
**Pull request by FenMei007S**
@tpmanley  Plz review code for LUMI water leak sensor

### CODE REVIEW

File: devicetypes/smartthings/zigbee-water-leak.src/zigbee-water-leak.groovy [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	def value = zs?.isAlarm1Set()?"wet":"dry"
+	[
+		name           : 'water',
+		value          : value,
+		descriptionText: "${device.displayName} is $value",
+		translatable   : true
+	]
+}
+
+def getBatteryPercentageResult(rawValue) {
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-water-leak.src/zigbee-water-leak.groovy]: Does this device support the battery percentage remaining attribute? The fingerprint doesn't include cluster 0x0001 which is the cluster that contains that attribute.

File: devicetypes/smartthings/zigbee-water-leak.src/zigbee-water-leak.groovy [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	log.debug "Refreshing Values"
+	def refreshCmds = []
+	refreshCmds += zigbee.enrollResponse()
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-water-leak.src/zigbee-water-leak.groovy]: The `enrollResponse` should be in `configure` only. `refresh` could be used to read the ZoneStatus attribute

File: devicetypes/smartthings/zigbee-water-leak.src/zigbee-water-leak.groovy [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	Map map = zigbee.getEvent(description)
+
+	if (!map) {
+		if (description?.startsWith('zone status')) {
+			map = getMoistureResult(description)
+		} else if(description?.startsWith('enroll request')){
+			List cmds = zigbee.enrollResponse()
+			log.debug "enroll response: ${cmds}"
+			result = cmds?.collect { new physicalgraph.device.HubAction(it) }
+		}else {
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-water-leak.src/zigbee-water-leak.groovy]: nit: should have a space before and after keywords which is missing here and after `if` in a few places

File: devicetypes/smartthings/zigbee-water-leak.src/zigbee-water-leak.groovy [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+	sendEvent(name: "checkInterval", value: 20 * 60 + 1 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	log.debug "Configuring Reporting"
+	                
+	return refresh() 
+}
+
+def getMoistureResult(description) {
+	ZoneStatus zs = zigbee.parseZoneStatus(description)
+	def value = zs?.isAlarm1Set()?"wet":"dry"
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-water-leak.src/zigbee-water-leak.groovy]: The `null` check shouldn't be required on `zs`.

File: devicetypes/smartthings/zigbee-water-leak.src/zigbee-water-leak.groovy [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+	if (!map) {
+		if (description?.startsWith('zone status')) {
+			map = getMoistureResult(description)
+		} else if(description?.startsWith('enroll request')){
+			List cmds = zigbee.enrollResponse()
+			log.debug "enroll response: ${cmds}"
+			result = cmds?.collect { new physicalgraph.device.HubAction(it) }
+		}else {
+			Map descMap = zigbee.parseDescriptionAsMap(description)
+            log.debug "${descMap}"
```
Comment (varzac) [inline devicetypes/smartthings/zigbee-water-leak.src/zigbee-water-leak.groovy]: You have some indentation issues

**Comment by FenMei007S** [comment]
@tpmanley @varzac  tom and zach, since lumi sensors just report battery low and normal by using 0500, discuessed with UI metedata developer, when report battery normal ,send out battery 50% , then UI will handle as normal, 5% for low.       but no raw data for 0500, only "zone status 0x0020--extended status 0x00"  how can we get the battery bit for here?

**Pull request closed without merging**